diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
index 07e903aa..b4e65ba3 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
@@ -626,7 +626,9 @@ JNIEXPORT void JNICALL Java_com_sun_glass_ui_ios_IosApplication__1initIDs
         return;
     haveIDs = 1;
 
-    assert(pthread_key_create(&GlassThreadDataKey, NULL) == 0);
+    // initializeThreadKeys
+    int res = pthread_key_create(&GlassThreadDataKey, NULL);
+    assert(res == 0);
 
     jApplicationClass = (*env)->NewGlobalRef(env, jClass);
     jApplicationReportException = (*env)->GetStaticMethodID(env, jClass, "reportException", "(Ljava/lang/Throwable;)V");
diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h b/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h
index d7862ebb..2424fd8a 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h
@@ -112,6 +112,7 @@ do {
     GlassThreadData *_GlassThreadData = (GlassThreadData*)pthread_getspecific(GlassThreadDataKey); \
     GLASS_POOL_EXIT
 
+#define MAT_IOS_DEBUG 1
 #if MAT_IOS_DEBUG
 
 #define GLASS_LOG(fmt, ...) NSLog((@"%s [line %d] " fmt), __PRETTY_FUNCTION__, __LINE__, ##__VA_ARGS__);
diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m
index 60a5255e..ac9cb784 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m
@@ -338,6 +338,18 @@ static EAGLContext * ctx = nil;
 
 // Called by the client whenever it draws (View.lock()->Pen.begin()->here)
 - (void)begin
+{
+    if ([[NSThread currentThread] isMainThread] == YES)
+    {
+        [self beginImpl];
+    }
+    else
+    {
+        [self performSelectorOnMainThread:@selector(beginImpl) withObject:nil waitUntilDone:YES];
+    }
+}
+
+- (void)beginImpl
 {
     // assert([EAGLContext currentContext] == clientContext);
     if ([EAGLContext currentContext] != clientContext) {
diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m
index 1c234817..3e43cab7 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m
@@ -836,9 +836,18 @@ jlong _1createWindow(JNIEnv *env, jobject jWindow, jlong jOwnerPtr, jlong jScree
 
             GLASS_LOG("SCREEN: %@", screen);
             CGRect applicationFrame = [screen bounds];
-            GLASS_LOG("FRAME: %@", applicationFrame);
+            GLASS_LOG("FRAME: %.1f %.1f", applicationFrame.size.width, applicationFrame.size.height);
 
-            mainWindow = [[GlassMainWindow alloc] initWithFrame:applicationFrame];
+            UIWindowScene *windowScene = (UIWindowScene *)[UIApplication sharedApplication].connectedScenes.allObjects.firstObject;
+            GLASS_LOG(@"WINDOW SCENE: %@", windowScene);
+            if (windowScene)
+            {
+                mainWindow = [[GlassMainWindow alloc] initWithWindowScene:windowScene];
+            }
+            else
+            {
+                mainWindow = [[GlassMainWindow alloc] initWithFrame:applicationFrame];
+            }
             mainWindowHost = [[GlassMainView alloc] initWithFrame:CGRectMake(0.0, 0.0, applicationFrame.size.width, applicationFrame.size.height)];
 
             // Set GlassViewController - responsible for orientation change, etc.
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c
index 5c822610..511fbb7c 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <math.h>
 
-#include "PrismES2Defs.h"
+#include "../PrismES2Defs.h"
 #include "com_sun_prism_es2_IOSGLContext.h"
 
 extern void printAndReleaseResources(jlong pf, jlong ctx, const char *message);
@@ -309,7 +309,7 @@ JNIEXPORT void JNICALL Java_com_sun_prism_es2_IOSGLContext_nMakeCurrent
     }
     interval = (vSyncNeeded) ? 1 : 0;
     ctxInfo->state.vSyncEnabled = vSyncNeeded;
-    setSwapInterval(ctxInfo->context, interval);
+    setSwapInterval((void *)(ctxInfo->context), interval);
     if (pulseLoggingRequested) {
         fprintf(stderr, "setSwapInterval(%d)\n", interval);
     }
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c
index 37a8f63a..3a54484d 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c
@@ -29,7 +29,7 @@
 #include <string.h>
 #include <math.h>
 
-#include "PrismES2Defs.h"
+#include "../PrismES2Defs.h"
 #include "com_sun_prism_es2_IOSGLDrawable.h"
 
 /*
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c
index 810c08c8..8f4d5be1 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <math.h>
 
-#include "PrismES2Defs.h"
+#include "../PrismES2Defs.h"
 #include "com_sun_prism_es2_IOSGLFactory.h"
 
 //Builtin library entrypoint
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m
index a29ba3f8..532989db 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m
@@ -60,7 +60,7 @@ void *getCurrentContext() {
     EAGLContext *ctx = [EAGLContext currentContext];
     // fprintf(stderr, "IOSWindowSystemInterface : getCurrentContext %x\n", ctx);
 
-    return ptr_to_jlong(ctx);
+    return (void *)ptr_to_jlong(ctx);
 }
 
 jboolean makeCurrentContext(void *context) {
