diff --git a/modules/javafx.graphics/src/main/native-prism-es2/PrismES2Defs.h b/modules/javafx.graphics/src/main/native-prism-es2/PrismES2Defs.h
index aa43478e350..bc728ce4aa0 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/PrismES2Defs.h
+++ b/modules/javafx.graphics/src/main/native-prism-es2/PrismES2Defs.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2026, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -69,16 +69,21 @@
 #ifdef __APPLE__
 #include <TargetConditionals.h>

-#define jlong_to_ptr(value) (intptr_t)(value)
-#define ptr_to_jlong(value) (jlong)((intptr_t)(value))
-
 #if TARGET_OS_IPHONE /* iOS */
+
+#define jlong_to_ptr(value) ((void*)((long)value))
+#define ptr_to_jlong(value) (jlong)((long)(value))
+
 #include <GL/gl.h>
 #include <GL/glext.h>
 #include "ios/ios-window-system.h"
 #define IS_EGL

 #elif TARGET_OS_MAC /* MacOSX */
+
+#define jlong_to_ptr(value) (intptr_t)(value)
+#define ptr_to_jlong(value) (jlong)((intptr_t)(value))
+
 /* Include the OpenGL headers */
 #include <GL/gl.h>
 #include <GL/glext.h>
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c
index 5c822610ad3..fe69cde4fc7 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLContext.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2026, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <math.h>

-#include "PrismES2Defs.h"
+#include "../PrismES2Defs.h"
 #include "com_sun_prism_es2_IOSGLContext.h"

 extern void printAndReleaseResources(jlong pf, jlong ctx, const char *message);
@@ -79,7 +79,6 @@ JNIEXPORT jlong JNICALL Java_com_sun_prism_es2_IOSGLContext_nInitialize
     jboolean result;
     ContextInfo *ctxInfo = NULL;
     DrawableInfo* dInfo =  (DrawableInfo* )jlong_to_ptr(nativeDInfo);
-    PixelFormatInfo* pfInfo =  (PixelFormatInfo* )jlong_to_ptr(nativePFInfo);
 
     if (dInfo == NULL) {
         return 0;
@@ -309,7 +308,7 @@ JNIEXPORT void JNICALL Java_com_sun_prism_es2_IOSGLContext_nMakeCurrent
     }
     interval = (vSyncNeeded) ? 1 : 0;
     ctxInfo->state.vSyncEnabled = vSyncNeeded;
-    setSwapInterval(ctxInfo->context, interval);
+    setSwapInterval((void *)(ctxInfo->context), interval);
     if (pulseLoggingRequested) {
         fprintf(stderr, "setSwapInterval(%d)\n", interval);
     }
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c
index 37a8f63aed1..1aead32f5a9 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLDrawable.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2026, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -29,7 +29,7 @@
 #include <string.h>
 #include <math.h>

-#include "PrismES2Defs.h"
+#include "../PrismES2Defs.h"
 #include "com_sun_prism_es2_IOSGLDrawable.h"

 /*
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c
index 810c08c85ba..98a75e1abfa 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSGLFactory.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2026, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -30,7 +30,7 @@
 #include <string.h>
 #include <math.h>

-#include "PrismES2Defs.h"
+#include "../PrismES2Defs.h"
 #include "com_sun_prism_es2_IOSGLFactory.h"

 //Builtin library entrypoint
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m
index a29ba3f8786..ca3ff57e5f1 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/IOSWindowSystemInterface.m
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2026, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -25,7 +25,7 @@

 #import <UIKit/UIKit.h>
 #import <dlfcn.h>
-#import "ios-window-system.h"
+#import "../PrismES2Defs.h"

 #if __IPHONE_OS_VERSION_MAX_ALLOWED >= 80000 // iOS 8.0 supported
 #include <OpenGLES/ES2/gl.h>
@@ -41,7 +41,7 @@ void deletePixelFormat(void* pixelFormat) {

 void *createContext(void *shareContext, void *view,
                     void *pixelFormat, int *viewNotReady) {
-    fprintf(stderr, "IOSWindowSystemInterface : share %x view %x pf % notready %\n",
+    fprintf(stderr, "IOSWindowSystemInterface : share %p view %p pf %p not ready %p\n",
             shareContext, view, pixelFormat, viewNotReady);

     EAGLContext *ctx = NULL;
@@ -60,7 +60,7 @@ void deletePixelFormat(void* pixelFormat) {
     EAGLContext *ctx = [EAGLContext currentContext];
     // fprintf(stderr, "IOSWindowSystemInterface : getCurrentContext %x\n", ctx);

-    return ptr_to_jlong(ctx);
+    return ctx;
 }

 jboolean makeCurrentContext(void *context) {
diff --git a/modules/javafx.graphics/src/main/native-prism-es2/ios/ios-window-system.h b/modules/javafx.graphics/src/main/native-prism-es2/ios/ios-window-system.h
index 4c6e6496268..e8d2d77bbe9 100644
--- a/modules/javafx.graphics/src/main/native-prism-es2/ios/ios-window-system.h
+++ b/modules/javafx.graphics/src/main/native-prism-es2/ios/ios-window-system.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2012, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2012, 2026, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -28,9 +28,6 @@
 */
 #import <jni.h>

-#define jlong_to_ptr(value) ((void*)((long)value))
-#define ptr_to_jlong(value) (jlong)((long)(value))
-
 void* createPixelFormat(jint* ivalues);
 void deletePixelFormat(void* pixelFormat);

