diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
index 07e903aa..b4e65ba3 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
@@ -626,7 +626,9 @@ JNIEXPORT void JNICALL Java_com_sun_glass_ui_ios_IosApplication__1initIDs
         return;
     haveIDs = 1;
 
-    assert(pthread_key_create(&GlassThreadDataKey, NULL) == 0);
+    // initializeThreadKeys
+    int res = pthread_key_create(&GlassThreadDataKey, NULL);
+    assert(res == 0);
 
     jApplicationClass = (*env)->NewGlobalRef(env, jClass);
     jApplicationReportException = (*env)->GetStaticMethodID(env, jClass, "reportException", "(Ljava/lang/Throwable;)V");
diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h b/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h
index d7862ebb..2424fd8a 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassMacros.h
@@ -112,6 +112,7 @@ do {
     GlassThreadData *_GlassThreadData = (GlassThreadData*)pthread_getspecific(GlassThreadDataKey); \
     GLASS_POOL_EXIT
 
+#define MAT_IOS_DEBUG 1
 #if MAT_IOS_DEBUG
 
 #define GLASS_LOG(fmt, ...) NSLog((@"%s [line %d] " fmt), __PRETTY_FUNCTION__, __LINE__, ##__VA_ARGS__);
diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m
index 60a5255e..ac9cb784 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassViewGL.m
@@ -338,6 +338,18 @@ static EAGLContext * ctx = nil;
 
 // Called by the client whenever it draws (View.lock()->Pen.begin()->here)
 - (void)begin
+{
+    if ([[NSThread currentThread] isMainThread] == YES)
+    {
+        [self beginImpl];
+    }
+    else
+    {
+        [self performSelectorOnMainThread:@selector(beginImpl) withObject:nil waitUntilDone:YES];
+    }
+}
+
+- (void)beginImpl
 {
     // assert([EAGLContext currentContext] == clientContext);
     if ([EAGLContext currentContext] != clientContext) {
diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m
index 1c234817..3e43cab7 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassWindow.m
@@ -836,9 +836,18 @@ jlong _1createWindow(JNIEnv *env, jobject jWindow, jlong jOwnerPtr, jlong jScree
 
             GLASS_LOG("SCREEN: %@", screen);
             CGRect applicationFrame = [screen bounds];
-            GLASS_LOG("FRAME: %@", applicationFrame);
+            GLASS_LOG("FRAME: %.1f %.1f", applicationFrame.size.width, applicationFrame.size.height);
 
-            mainWindow = [[GlassMainWindow alloc] initWithFrame:applicationFrame];
+            UIWindowScene *windowScene = (UIWindowScene *)[UIApplication sharedApplication].connectedScenes.allObjects.firstObject;
+            GLASS_LOG(@"WINDOW SCENE: %@", windowScene);
+            if (windowScene)
+            {
+                mainWindow = [[GlassMainWindow alloc] initWithWindowScene:windowScene];
+            }
+            else
+            {
+                mainWindow = [[GlassMainWindow alloc] initWithFrame:applicationFrame];
+            }
             mainWindowHost = [[GlassMainView alloc] initWithFrame:CGRectMake(0.0, 0.0, applicationFrame.size.width, applicationFrame.size.height)];
 
             // Set GlassViewController - responsible for orientation change, etc.
