diff --git a/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m b/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
index 07e903aa..b4e65ba3 100644
--- a/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
+++ b/modules/javafx.graphics/src/main/native-glass/ios/GlassApplication.m
@@ -626,7 +626,9 @@ JNIEXPORT void JNICALL Java_com_sun_glass_ui_ios_IosApplication__1initIDs
         return;
     haveIDs = 1;
 
-    assert(pthread_key_create(&GlassThreadDataKey, NULL) == 0);
+    // initializeThreadKeys
+    int res = pthread_key_create(&GlassThreadDataKey, NULL);
+    assert(res == 0);
 
     jApplicationClass = (*env)->NewGlobalRef(env, jClass);
     jApplicationReportException = (*env)->GetStaticMethodID(env, jClass, "reportException", "(Ljava/lang/Throwable;)V");
