diff --git a/make/modules/javafx.graphics/Lib.gmk b/make/modules/javafx.graphics/Lib.gmk
index 842f42147..60e797435 100644
--- a/make/modules/javafx.graphics/Lib.gmk
+++ b/make/modules/javafx.graphics/Lib.gmk
@@ -21,7 +21,7 @@ $(eval $(call SetupJdkLibrary, BUILD_LIBPRISM_COMMON, \

 TARGETS += $(BUILD_LIBPRISM_COMMON)

-ifeq ($(call isTargetOs, linux macosx), true)
+ifeq ($(call isTargetOs, linux macosx ios), true)
 ################################################################################
 ## Build libprism_es2
 ################################################################################
@@ -33,16 +33,21 @@ endif
 ifeq ($(call isTargetOs, macosx), true)
    PLATFORM_SRC := $(SRC_SHARE_PRISM_ES2)/macosx
 endif
+ifeq ($(call isTargetOs, ios), true)
+   PLATFORM_SRC := $(SRC_SHARE_PRISM_ES2)/ios
+endif
 $(eval $(call SetupJdkLibrary, BUILD_LIBPRISM_ES2, \
     NAME := prism_es2, \
     OPTIMIZATION := HIGH, \
     LIBS_linux := -lX11 -lXxf86vm -lGL, \
     LIBS_macosx := -framework AppKit -framework ApplicationServices -framework Carbon -framework OpenGL -framework QuartzCore -framework Security, \
+    LIBS_ios := -framework AppKit -framework ApplicationServices -framework Carbon -framework OpenGL -framework QuartzCore -framework Security, \
     SRC := $(PLATFORM_SRC), \
     EXTRA_FILES := $(SRC_SHARE_PRISM_ES2)/GLContext.c, \
     EXTRA_FILES += $(SRC_SHARE_PRISM_ES2)/GLFactory.c, \
     EXTRA_FILES += $(SRC_SHARE_PRISM_ES2)/GLPixelFormat.c, \
     CFLAGS_macosx := -I$(SRC_SHARE_PRISM_ES2), \
+    CFLAGS_ios := -I$(SRC_SHARE_PRISM_ES2), \
     CFLAGS := -Wno-error, \
 ))

@@ -81,13 +86,16 @@ endif
 ifeq ($(call isTargetOs, windows), true)
    PLATFORM_GLASS_SRC := $(SRC_SHARE_GLASS)/win
 endif
-
+ifeq ($(call isTargetOs, ios), true)
+   PLATFORM_GLASS_SRC := $(SRC_SHARE_GLASS)/ios
+endif
 $(eval $(call SetupJdkLibrary, BUILD_LIBGLASS, \
     NAME := glass, \
     OPTIMIZATION := HIGH, \
     SRC := $(PLATFORM_GLASS_SRC), \
     LIBS_linux := -lX11, \
     LIBS_macosx := -framework AppKit -framework ApplicationServices -framework Metal -framework Carbon -framework OpenGL -framework QuartzCore -framework Security -framework Network, \
+    LIBS_ios := -framework AppKit -framework ApplicationServices -framework Metal -framework Carbon -framework OpenGL -framework QuartzCore -framework Security -framework Network, \
     LIBS_windows := -manifest -dynamicbase -nxcompat delayimp.lib gdi32.lib urlmon.lib Comdlg32.lib winmm.lib imm32.lib shell32.lib Uiautomationcore.lib dwmapi.lib version.lib /DELAYLOAD:user32.dll /DELAYLOAD:urlmon.dll /DELAYLOAD:winmm.dll /DELAYLOAD:shell32.dll /DELAYLOAD:Uiautomationcore.dll /DELAYLOAD:dwmapi.dll /DELAYLOAD:version.dll, \
     EXTRA_FILES := $(EXTRA_GLASS_SRC), \
     CFLAGS_macosx := -DMACOS_MIN_VERSION_MAJOR=11 -DMACOS_MIN_VERSION_MINOR=0, \
@@ -173,11 +181,15 @@ endif
 ## Build libjavafx_iio
 ################################################################################

+ifeq ($(call isTargetOs, ios), true)
+   PLATFORM_IIO_SRC := $(OPENJFX_MODULES_SRC)/modules/javafx.graphics/src/main/native-iio/ios
+endif
 $(eval $(call SetupJdkLibrary, BUILD_LIBJAVAFXIIO, \
     NAME := javafx_iio, \
     LINK_TYPE := C++, \
     OPTIMIZATION := HIGH, \
     SRC := $(OPENJFX_MODULES_SRC)/modules/javafx.graphics/src/main/native-iio/libjpeg, \
+    EXTRA_SRC := $(PLATFORM_IIO_SRC), \
     EXTRA_FILES := $(OPENJFX_MODULES_SRC)/modules/javafx.graphics/src/main/native-iio/jpegloader.c, \
     LIBS_windows := -manifest -dynamicbase -nxcompat, \
     CFLAGS_windows := -EHsc -D_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR -DINLINE=__inline -permissive, \
